libr/include/r_util/r_str.h

```diff
@@ -19,6 +19,51 @@ typedef enum {
 	R_STRING_ENC_GUESS = 'g',
 } RStrEnc;
 
+// r_strf API can be used for short strings that are needed for a short period
```

Comment:
please note that, of course, we cannot really check how much memory is already used in the stack. Again, I want to highlight how this thing should be used with care! I think `sdb_fmt` is already used on strings that are not of a known size. Whatever may come from the user or from the binary should not be passed to `r_strf`. `r_strf` is ideal for converting numbers into strings or for creating concatenating small pieces of strings that are all hardcoded (e.g. `r_meta_type_to_string()`, etc.), but not to work with random stuff like `section->name` or `symbol->name` or things like that, that could be manipulated by the binary/user.

In existing `sdb_fmt` the problem, instead, is that those strings may be easily truncated.

Reply:
well its a problem that the programmer must cope with it and in 10 years of using sdb this never happened

Reply:
but i agree that it should be addressed properly with a runtime assert

Reply:
As said, `please note that, of course, we cannot really check how much memory is already used in the stack`. So there's no real check we can do here, apart from what's already done.